### PR TITLE
KAFKA-1901: follow-up on KAFKA-1901; Added error handling.

### PR DESCRIPTION
**Pull request by omkreddy**
@guozhangwang  added .git/refs/heads/ file existence check.

### CODE REVIEW

**Comment by guozhangwang** [comment]
@omkreddy If there is no file exist in .git/refs/heads, could we search in .git/packed-refs?

**Comment by omkreddy** [comment]
@guozhangwang  Yes, we can implement. But there may be other git scenarios, we may get similar exception. Current patch captures git commitID, if we are building on trunk/branch/tag (these common ways of building, for other scenarios it will just returns "unknown" commitID.  I feel current functionality is sufficient for  general build scenarios.  Let me know your thoughts.

File: build.gradle [hunk 1]
```diff
@@ -404,7 +404,11 @@ project(':clients') {
       def headRef = file("../.git/HEAD").text
       if (headRef.contains('ref: ')) {
         headRef = headRef.replaceAll('ref: ', '').trim()
-        commitId = file("../.git/$headRef").text.trim().take(takeFromHash)
+        if (file("../.git/$headRef").exists()) {
+          commitId = file("../.git/$headRef").text.trim().take(takeFromHash)
+        } else {
+          commitId
```
Comment (ijuma) [inline build.gradle]: Are we supposed to assign a value to `commitId` in the `else` or am I missing something? Or maybe we don't need the `else` at all?

**Comment by omkreddy** [comment]
@ijuma thank for the review. I just removed unnecessary else block.

**Comment by ijuma** [comment]
@guozhangwang Do you want to check this PR again?

**Comment by guozhangwang** [comment]
LGTM. Merging to trunk.

**Pull request closed without merging**